### ISSUE TITLE
[Bug]: Connecting to ESP devices with WiFi over 2.4.1 causes the device to go into a bootloop

### ISSUE THREAD

**Issue by GMart** [meshtastic/Meshtastic-Android#1264]
### Contact Details

@droidbot2

### What happened?

This does not happen if I directly connect to the IP address. If I use client.meshtastic.org, it hangs. 
If I use the Android app, the device immediately reboots, and will continue to reboot as long as the app is trying to connect to the device.

### App Version

2.4.11 or 2.4.12

### Phone

Galaxy S 21+, Android 14

### Device

Any: I have tried Heltec VME290, E213, and Wireless paper, to name a few. However 2.4.3 works fine on the Heltec Wireless Stick Lite V3.

### Firmware

2.4.3 or 2.5.0

### Relevant log output

```shell
INFO | ??:??:?? 13 [RangeTestModule] Range Test Module - Disabled

INFO | ??:??:?? 13 [WifiConnect] Reconnecting to WiFi access point UnitG4

INFO | ??:??:?? 18 [Screen] Done with boot screen...

DEBUG | ??:??:?? 18 [Screen] showing standard frames

DEBUG | ??:??:?? 18 [Screen] Showing 0 module frames

DEBUG | ??:??:?? 18 [Screen] Total frame count: 103

DEBUG | ??:??:?? 18 [Screen] Added modules. numframes: 0

DEBUG | ??:??:?? 18 [Screen] Finished building frames. numframes: 5

DEBUG | ??:??:?? 18 [Screen] determineMode(): refresh=FULL, reason=FLAGGED_COSMETIC, frameFlags=0x7

DEBUG | ??:??:?? 18 [Screen] Updating E-Paper... done

DEBUG | ??:??:?? 18 [Screen] Async full-refresh begins (dropping frames)

DEBUG | ??:??:?? 18 [Screen] refresh=SKIPPED, reason=EXCEEDED_RATELIMIT_FAST, frameFlags=0x3

DEBUG | ??:??:?? 18 WiFi-Event 4: Connected to access point

DEBUG | ??:??:?? 20 WiFi-Event 7: Obtained IP address: 192.168.0.40

INFO | ??:??:?? 20 Starting WiFi network services

INFO | ??:??:?? 20 mDNS responder started

INFO | ??:??:?? 20 mDNS Host: Meshtastic.local

INFO | ??:??:?? 20 Starting NTP time client

DEBUG | ??:??:?? 20 Initializing Web Server ...

INFO | ??:??:?? 20 Starting Secure Web Server...

INFO | ??:??:?? 20 Starting Insecure Web Server...

INFO | ??:??:?? 20 Web Servers Ready! :-)

INFO | ??:??:?? 20 API server listening on TCP port 4403

DEBUG | ??:??:?? 20 [WifiConnect] Updating NTP time from meshtastic.pool.ntp.org

DEBUG | ??:??:?? 20 [WifiConnect] NTP Request Success - Setting RTCQualityNTP if needed

DEBUG | ??:??:?? 20 [WifiConnect] Upgrading time to quality NTP

DEBUG | 14:24:15 20 [WifiConnect] Read RTC time as 1726842255

DEBUG | 14:24:16 21 [EInkDynamicDisplay] Async full-refresh complete

DEBUG | 14:24:17 22 [Screen] determineMode(): refresh=FAST, reason=NO_OBJECTIONS, fastRefreshCount=0, frameFlags=0x3

DEBUG | 14:24:17 22 [Screen] Updating E-Paper... done
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by andrekir** [comment]
I can't reproduce this. any chance you can grab the serial logs showing a reboot? does reverting to an older release of the Android app resolve the issue?

**Comment by GMart** [comment]
I can try again, but that serial log is what I get. The log just stops when I connect via Wifi. I can try an older release of Android, will try later.

**Comment by todd-herbert** [comment]
Better to test this with Wireless Paper than VME290, if possible. Wireless Paper has a separate USB serial IC, but VME290 is using the native ESP32 serial, so if there's an error, it just dies before it has time to print debugging info.

**Comment by GMart** [comment]
It's fixed with 2.5.2 Android app. Not sure what changed. Device is Heltec E290 running 2.5.6.
